dhcp-relay/dhcp_kern_xdp.c

```diff
@@ -256,15 +345,19 @@ int xdp_dhcp_relay(struct xdp_md *ctx) {
 		goto out;
 
 	/* Handle DHCP packets only */
-	if (udp->dest != bpf_htons(DHCP_SERVER_PORT) && udp->dest != bpf_htons(DHCP_CLIENT_PORT))
+	if (udp->dest != bpf_htons(DHCP_SERVER_PORT) && udp->dest != bpf_htons(DHCP_CLIENT_PORT)) {
+		bpf_printk("Not DHCP");
 		goto out;
+	}
 
 	/* Increase IP length header */
 	ip->tot_len += bpf_htons(delta);
 
 	/* Increase UDP length header */
 	udp->len += bpf_htons(delta);
 
+	udp->check = 0;
```

Comment:
Here you're modifying the packet, but any `goto out` statements below will end up returning `XDP_PASS`, passing up a potentially partially-modified packet. `rc` should be changed to `XDP_DROP` at the point of the first packet modification.